tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsWriterTests.cs

```diff
@@ -0,0 +1,144 @@
+﻿// <copyright file="DataStreamsWriterTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.DataStreamsMonitoring;
+using Datadog.Trace.DataStreamsMonitoring.Aggregation;
+using Datadog.Trace.DataStreamsMonitoring.Hashes;
+using Datadog.Trace.DataStreamsMonitoring.Transport;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Util;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.DataStreamsMonitoring;
+
+public class DataStreamsWriterTests
+{
+    private const int BucketDurationMs = 1_000; // 1 second
+
+    [Fact]
+    public async Task DoesNotWriteIfNoStats_Periodic()
+    {
+        var bucketDurationMs = 100; // 100 ms
+        var api = new StubApi();
+        var writer = CreateWriter(api, bucketDurationMs * 1_000_000);
+
+        await Task.Delay(bucketDurationMs * 10);
+
+        api.Sent.Should().BeEmpty();
+    }
+
+    [Fact]
+    public async Task DoesNotWriteIfNoStats_OnClose()
+    {
+        var bucketDuration = 100_000_000; // 100 ms
+        var api = new StubApi();
+        var writer = CreateWriter(api, bucketDuration);
+
+        await writer.DisposeAsync();
+
+        api.Sent.Should().BeEmpty();
+    }
+
+    [Fact]
+    public async Task WritesAStatsPointAfterDelay()
+    {
+        var bucketDurationMs = 100; // 100 ms
+        var api = new StubApi();
+        var writer = CreateWriter(api, bucketDurationMs * 1_000_000);
+
+        writer.Add(CreateStatsPoint());
+
+        await Task.Delay(bucketDurationMs * 10);
+
+        await writer.DisposeAsync();
```

Comment:
Could you read the `api.Sent` value before calling `DisposeAsync` just to make sure the behavior of `DisposeAsync` doesn't impact this test?

Reply:
Yeah, I should do.... hope that doesn't cause any flake 😄